docs/reference/scripting/using.asciidoc

```diff
@@ -197,6 +197,86 @@ them in other cases to remove ambiguity.
 Use this abbreviated syntax anywhere that {es} supports scripts, such as
 when you're creating <<runtime-mapping-fields,runtime fields>>.
 
+[discrete]
+[[script-fields-api]]
+=== Access document fields
+The examples from previous sections use the `field` API to access document
+fields:
+
+[source,painless]
+----
+field('my_field').get(null)
+----
+
+The `field` API fundamentally changes how you access documents in Painless.
+Previously, you had to declare `doc` and the field name that you wanted to
```

Comment:
TODO(stu/jack) write an explanation of `doc` as multi-valued, containing a key per mapping.